stan/math/prim/mat/meta/is_eigen.hpp

```diff
@@ -43,11 +44,37 @@ template <typename T, int R, int C>
 struct is_eigen_matrix_impl<Eigen::Matrix<T, R, C>> : std::true_type {};
 template <typename T>
 struct is_eigen_matrix_impl<Eigen::SparseMatrix<T>> : std::true_type {};
-
 }  // namespace internal
 
 template <typename T>
 struct is_eigen_matrix : internal::is_eigen_matrix_impl<std::decay_t<T>> {};
 
+namespace internal {
+template <typename T>
+struct is_eigen_array_impl : std::false_type {};
+template <typename T, int R, int C>
+struct is_eigen_array_impl<Eigen::Array<T, R, C>> : std::true_type {};
+}  // namespace internal
+
+template <typename T>
+struct is_eigen_array : internal::is_eigen_array_impl<std::decay_t<T>> {};
+
+template <typename T>
+using is_eigen_matrix_or_array
+    = math::disjunction<is_eigen_matrix<T>, is_eigen_array<T>>;
+
+namespace internal {
+template <typename T>
+struct is_eigen_contiguous_map_impl : std::false_type {};
+template <typename T, int Opts>
```

Comment:
Could this just be variadic?

Reply:
We should probs do that for all of these that can take in multiple values

Reply:
I think I don't understand the question. Can you give an example?